bug: pitch control warning scoped to single wing instead of whole aeroplane

## Bug Report

### Description
The "No pitch control surface assigned" warning in the AeroplaneTree is evaluated per-wing. A wing that only has ailerons shows the warning even when another wing on the same aeroplane has an elevator assigned.

The check should be aeroplane-wide: if *any* wing has a pitch control surface (elevator, elevon, stabilator, ruddervator), no wing should show the warning.

### Steps to Reproduce
1. Create an aeroplane with two wings
2. Assign an elevator to a TED on wing 1
3. Assign an aileron to a TED on wing 2
4. Wing 2 shows: "No pitch control surface assigned. Auto-trim and stability analysis require an elevator, elevon, or stabilator."

### Expected Behavior
No warning on any wing, because the aeroplane has an elevator on wing 1.

### Current Behavior
Each wing checks only its own TEDs for pitch roles. Wings without a pitch surface show the warning regardless of other wings.

### Root Cause
`hasPitchSurface` in `AeroplaneTree.tsx:802-810` iterates only over the current wing's `x_secs`. It needs to check all wings on the aeroplane.

```tsx
// Current (per-wing)
const hasPitchSurface = useMemo(() => {
    if (!wing) return true;
    return wing.x_secs.some((xsec) => {
      const ted = getTedData(xsec);
      if (!ted) return false;
      const role = (ted.role as string) ?? "";
      return PITCH_ROLES.has(role);
    });
  }, [wing]);
```

### Fix
Check across all wings from the aeroplane data (available via `useAeroplane` or the existing `wingNames` / aeroplane context in the component).

### Files
- `frontend/components/workbench/AeroplaneTree.tsx:79` — `PITCH_ROLES` set
- `frontend/components/workbench/AeroplaneTree.tsx:802-810` — `hasPitchSurface` check
- `frontend/components/workbench/AeroplaneTree.tsx:892-898` — warning render
- `frontend/__tests__/AeroplaneTreeRoleIcons.test.tsx:164-220` — tests need updating

## 🏷️ has-reproduction

> Label \`has-reproduction\` added by **supercycle/bug** · 2026-05-12

---

### Failing tests (RED)

\`frontend/__tests__/AeroplaneTreeRoleIcons.test.tsx\`:

1. \`AeroplaneTree pitch warning — aeroplane-wide scope (gh-464) > hides warning on aileron-only wing when another wing has elevator\`
2. \`AeroplaneTree pitch warning — aeroplane-wide scope (gh-464) > hides warning on rudder-only wing when another wing has elevon\`

### Failing output

\`\`\`
FAIL  __tests__/AeroplaneTreeRoleIcons.test.tsx > AeroplaneTree pitch warning — aeroplane-wide scope (gh-464) > hides warning on aileron-only wing when another wing has elevator
AssertionError: expected <div role="alert" …(1)></div> to be null

- Expected:
null

+ Received:
<div
  class="mt-2 rounded-lg border border-amber-500/30 bg-amber-500/10 px-3 py-2 text-[11px] text-amber-400"
  role="alert"
>
  No pitch control surface assigned. Auto-trim and stability analysis require an elevator, elevon, or stabilator.
</div>

 Test Files  1 failed (1)
      Tests  2 failed | 10 passed (12)
\`\`\`

### Test scenarios

| Test | wing.x_secs (selected) | all wings | expected |
|---|---|---|---|
| aileron + elevator | aileron only | aileron wing + elevator wing | no warning |
| rudder + elevon | rudder only | elevon wing + rudder wing | no warning |
| no pitch anywhere | aileron only | aileron wing + flap wing | warning |

The 3rd test (no pitch anywhere) passes already because per-wing and aeroplane-wide checks agree when no pitch surface exists at all. The first two **only pass after the aeroplane-wide fix**.

--- a/frontend/__tests__/AeroplaneTreeRoleIcons.test.tsx
+++ b/frontend/__tests__/AeroplaneTreeRoleIcons.test.tsx
@@ -1,7 +1,7 @@
 /**
  * Tests for AeroplaneTree role icons and pitch control surface warning (gh-450).
  */
-import { describe, it, expect, vi } from "vitest";
+import { describe, it, expect, vi, beforeEach } from "vitest";
 import { render, screen } from "@testing-library/react";
 import React from "react";
 import type { XSec } from "@/hooks/useWings";
@@ -57,12 +57,26 @@ vi.mock("@/components/workbench/AeroplaneContext", () => ({
   }),
 }));
 
-let mockWingData: { name: string; symmetric: boolean; x_secs: XSec[]; design_model?: string } | null = null;
+type MockWing = { name: string; symmetric: boolean; x_secs: XSec[]; design_model?: string };
+let mockWingData: MockWing | null = null;
+let mockAllWingsData: MockWing[] = [];
+
+function resolveMockAllWings(): MockWing[] {
+  if (mockAllWingsData.length > 0) return mockAllWingsData;
+  if (mockWingData) return [mockWingData];
+  return [];
+}
 
 vi.mock("@/hooks/useWings", () => ({
   useWing: () => ({
     wing: mockWingData,
     isLoading: false,
+    mutate: vi.fn(),
+  }),
+  useAllWingData: () => ({
+    wings: resolveMockAllWings(),
+    isLoading: false,
+    error: undefined,
     mutate: vi.fn(),
   }),
 }));
@@ -161,6 +175,10 @@ describe("AeroplaneTree role icons (gh-450)", () => {
 });
 
 describe("AeroplaneTree pitch warning (gh-450)", () => {
+  beforeEach(() => {
+    mockAllWingsData = [];
+  });
+
   it("shows warning when wing has no pitch control surface", () => {
     mockWingData = makeWing([
       makeXsec({ trailing_edge_device: { role: "aileron", label: "" } }),
@@ -219,5 +237,116 @@ describe("AeroplaneTree pitch warning (gh-450)", () => {
     expect(
       screen.queryByText(/No pitch control surface assigned/),
     ).toBeNull();
+  });
+});
+
+describe("AeroplaneTree pitch warning — aeroplane-wide scope (gh-464)", () => {
+  // Why: the warning must reflect aeroplane-wide capability, not just the
+  // currently-selected wing. A wing of ailerons should not show the warning
+  // when another wing on the same aeroplane has an elevator.
+  beforeEach(() => {
+    mockAllWingsData = [];
+  });
+
+  it("hides warning on aileron-only wing when another wing has elevator", () => {
+    const aileronWing: MockWing = {
+      name: "Main Wing",
+      symmetric: true,
+      x_secs: [
+        makeXsec({ trailing_edge_device: { role: "aileron", label: "" } }),
+        makeXsec(),
+      ],
+      design_model: "wc",
+    };
+    const tailWing: MockWing = {
+      name: "Horizontal Tail",
+      symmetric: true,
+      x_secs: [
+        makeXsec({ trailing_edge_device: { role: "elevator", label: "" } }),
+        makeXsec(),
+      ],
+      design_model: "wc",
+    };
+    mockWingData = aileronWing;
+    mockAllWingsData = [aileronWing, tailWing];
+
+    render(
+      <AeroplaneTree
+        aeroplaneId="1"
+        wingNames={["Main Wing", "Horizontal Tail"]}
+        aeroplaneName="Test Plane"
+      />,
+    );
+    expect(
+      screen.queryByText(/No pitch control surface assigned/),
+    ).toBeNull();
+  });
+
+  it("hides warning on rudder-only wing when another wing has elevon", () => {
+    const rudderWing: MockWing = {
+      name: "Vertical Tail",
+      symmetric: false,
+      x_secs: [
+        makeXsec({ trailing_edge_device: { role: "rudder", label: "" } }),
+        makeXsec(),
+      ],
+      design_model: "wc",
+    };
+    const flyingWing: MockWing = {
+      name: "Main Wing",
+      symmetric: true,
+      x_secs: [
+        makeXsec({ trailing_edge_device: { role: "elevon", label: "" } }),
+        makeXsec(),
+      ],
+      design_model: "wc",
+    };
+    mockWingData = rudderWing;
+    mockAllWingsData = [flyingWing, rudderWing];
+
+    render(
+      <AeroplaneTree
+        aeroplaneId="1"
+        wingNames={["Main Wing", "Vertical Tail"]}
+        aeroplaneName="Test Plane"
+      />,
+    );
+    expect(
+      screen.queryByText(/No pitch control surface assigned/),
+    ).toBeNull();
+  });
+
+  it("still shows warning when no wing on the aeroplane has a pitch surface", () => {
+    const aileronWing: MockWing = {
+      name: "Main Wing",
+      symmetric: true,
+      x_secs: [
+        makeXsec({ trailing_edge_device: { role: "aileron", label: "" } }),
+        makeXsec(),
+      ],
+      design_model: "wc",
+    };
+    const flapWing: MockWing = {
+      name: "Inner Wing",
+      symmetric: true,
+      x_secs: [
+        makeXsec({ trailing_edge_device: { role: "flap", label: "" } }),
+        makeXsec(),
+      ],
+      design_model: "wc",
+    };
+    mockWingData = aileronWing;
+    mockAllWingsData = [aileronWing, flapWing];
+
+    render(
+      <AeroplaneTree
+        aeroplaneId="1"
+        wingNames={["Main Wing", "Inner Wing"]}
+        aeroplaneName="Test Plane"
+      />,
+    );
+    expect(
+      screen.getByText(/No pitch control surface assigned/),
+    ).toBeTruthy();
   });
 });

--- a/frontend/__tests__/FuselageTree.test.tsx
+++ b/frontend/__tests__/FuselageTree.test.tsx
@@ -21,6 +21,7 @@ vi.mock("lucide-react", () => {
 
 vi.mock("@/hooks/useWings", () => ({
   useWing: () => ({ wing: null, isLoading: false, mutate: vi.fn() }),
+  useAllWingData: () => ({ wings: [], isLoading: false, error: undefined, mutate: vi.fn() }),
 }));
 
 const mockFuselageData = {

--- a/frontend/__tests__/XsecTreeCrud.test.tsx
+++ b/frontend/__tests__/XsecTreeCrud.test.tsx
@@ -32,6 +32,7 @@ const mockWing = {
 
 vi.mock("@/hooks/useWings", () => ({
   useWing: () => ({ wing: mockWing, isLoading: false, mutate: vi.fn() }),
+  useAllWingData: () => ({ wings: [mockWing], isLoading: false, error: undefined, mutate: vi.fn() }),
 }));
 
 vi.mock("@/hooks/useWingConfig", () => ({

--- a/frontend/components/workbench/AeroplaneTree.tsx
+++ b/frontend/components/workbench/AeroplaneTree.tsx
@@ -3,7 +3,7 @@
 import { useState, useEffect, useMemo } from "react";
 import { ChevronDown, ChevronRight, Plus, Trash2, Eye, EyeOff, Loader, PanelLeftClose, Pencil } from "lucide-react";
 import { useAeroplaneContext } from "@/components/workbench/AeroplaneContext";
-import { useWing } from "@/hooks/useWings";
+import { useWing, useAllWingData } from "@/hooks/useWings";
 import type { XSec } from "@/hooks/useWings";
 import { useWingConfig } from "@/hooks/useWingConfig";
 import { API_BASE } from "@/lib/fetcher";
@@ -799,15 +799,21 @@ export function AeroplaneTree(props: Readonly<AeroplaneTreeProps>) {
     onNodeEdit, onEditSpar, onDeleteSpar, onEditTed, onDeleteTed, onAddSpar, onAddTed,
   });
 
+  // Pitch-control capability is an aeroplane-wide property: any wing with an
+  // elevator/elevon/stabilator satisfies the trim/stability requirement, so
+  // the warning must check across all wings rather than just the selected one.
+  const { wings: allWings } = useAllWingData(aeroplaneId, wingNames);
   const hasPitchSurface = useMemo(() => {
-    if (!wing) return true;
-    return wing.x_secs.some((xsec) => {
-      const ted = getTedData(xsec);
-      if (!ted) return false;
-      const role = (ted.role as string) ?? "";
-      return PITCH_ROLES.has(role);
-    });
-  }, [wing]);
+    if (allWings.length === 0) return true;
+    return allWings.some((w) =>
+      w.x_secs.some((xsec) => {
+        const ted = getTedData(xsec);
+        if (!ted) return false;
+        const role = (ted.role as string) ?? "";
+        return PITCH_ROLES.has(role);
+      }),
+    );
+  }, [allWings]);
 
   // Build tree data
   const treeData = buildTreeData({

